### DISCUSSION TITLE
Combining form fields into one control - validation issues

### DISCUSSION THREAD

**Discussion by npup** [react-hook-form/react-hook-form#9638] [Q&A] [answered]
Hi all,

I have a requirement of handling business objects in form contexts without re-mapping their properties in any way.

Specifically, the case I try to solve right now involves editing points in time, stored in such objects.  The objects all store these points in time as two separate string properties: date and time, respectively.

For example:

```
{
    startDate: "2022-12-10",
    startTime: "13:37",
    ...
}
```
Using a MUI date picker - which works with `Date` instances (not strings), I first created a component that wraps a that picker, and handles transformation between `string` and `Date` instances.  This works fine afaikt.

The next step was to create a component that can abstract away the combining of date+time strings and feed it to the working component mentioned just above.  This also works fine, in many ways.  The technique is to create a standalone control for the full (combined) value, `watch` it for changes, and then pass the parts of it to the main context via `setValue(...)`.

**The problem** I see is that the way I wrap the "text/Date-aware" datepicker, the `onSubmit` validation does not get a chance to trigger.  **My question** is if you can suggest any solution to this.  I have tried to tap into the form contexts involved and see if it can be linked somehow, but I haven't succeeded via any API known to me (yet).

Any suggestions for changes to my approach - or a different approach altogether - would be very interesting and highly appreciated.

Here is a codesandbox with an example of what I just described (just noticed it didn't load correctly when i revisited it due to some dependency problem - but a reload fixed it.  Network fluke, I suppose):

https://codesandbox.io/s/trusting-gianmarco-uvehdl?file=/src/components/DateTimePicker.jsx

Best regards
/p

**Comment by leapful** [comment, marked as answer]
@npup 

Your wrapper DateTimePicker component did not trigger validation for themself as an inner form context and also the default value in case `dateStr` and `timeStr` is empty will be a blank space which is valid with rule "required".

Just trigger inner validation onSubmit or onError of parent outer form and update correct default value will solve the issue.

Working demo: https://codesandbox.io/s/bold-currying-olpv3w?file=/src/components/DateTimePicker.jsx

**Reply by npup** [reply]
Thank you so much for helping me (with an example even), and for explaining it so well. Maybe someone else can learn from this too!
